src/wallet/wallet.cpp

```diff
@@ -3726,6 +3765,10 @@ void CWallet::GetFilteredNotes(
                 continue;
             }
 
+            if (IsLockedNote(jsop.hash, jsop.js, jsop.n)) {
```

Comment:
Add a comment here e.g. // Skip locked notes.

Also rebase on master, as z_listunspent PR has been merged, and you'll need to do the same thing here in GetUnspentFilteredNotes.